### PR TITLE
cartopy feature download

### LINKED ISSUE #4302: ABF file format reference
## 📰 Custom Issue

Link checks are currently failing all over the PRs because this link is timing out: http://cliveg.bu.edu/modismisr/lai3g-fpar3g.html.  This link is given as the file format reference within the [iris.fileformats.abf](https://github.com/SciTools/iris/blob/main/lib/iris/fileformats/abf.py) module.

Using the Wayback Machine, we can see [what the page looked like in February](http://web.archive.org/web/20200218234926/http://cliveg.bu.edu/modismisr/lai3g-fpar3g.html) but it appears to be describing a data set, rather than a file format.  According to https://github.com/SciTools/iris/pull/630#issuecomment-21776938, the format description had already disappeared from this page in 2013, and @bblay helpfully provided a string to google, which took me to [this document on AWS](http://nasanex.s3.amazonaws.com/AVHRR/GIMMS/LAI3G/00README_V01.pdf).

Not sure what the best fix would be.  Update the link to the AWS link?  Just remove the link?

### PR DESCRIPTION
**Pull request by bjlittle**
### Description
This PR fixes the broken ABF fileformat documentation URL link.

Closes #4302

---
[Consult Iris pull request check list]( https://scitools-iris.readthedocs.io/en/latest/developers_guide/contributing_pull_request_checklist.html)

### CODE REVIEW

**Review by rcomer — APPROVED** [review]
LGTM.  Just waiting for the CI to do its thing.

**Review by pp-mo — APPROVED** [review]
Makes sense.
I wondered whether the link should live in `docs/src/common_links.inc`, 
but I think that is unhelpful because it is needed in code-docstrings, not just docs (rst).

**Comment by rcomer** [comment]
Looks like the Natural Earth server is having issues again 😞

**Comment by bjlittle** [comment]
Well... this PR took an unexpected diversion :rofl:

**Comment by rcomer** [comment]
@bjlittle maybe update the title?  I’m not sure I’m qualified to review this anymore.

File: .cirrus.yml
Comment (bjlittle) [inline .cirrus.yml:29]: We need to nudge the build number in order to force the `cartopy_cache` miss, and a subsequent download to populate the cache.

**Comment by rcomer** [comment]
> This is actually a pretty handy little general utility, and whatsmore, as with cartopy it **won't be bundled with iris** when it is packaged.

> I know that populating the cartopy cache is a pain-point for many users, particularly when pre-caching for offline usage, or when attempting to develop behind a firewall with no internet access e.g., on a secure HPC, so this may somewhat help ease the burden i.e., you can use the utility to download a cartopy cache, then lift and shift it to your installation, or be confident that you have all the resource you need before working offline.

I confused here: if the script isn't bundled with Iris, how do the users take advantage of it?

**Comment by bjlittle** [comment]
@rcomer I've followed the same pattern as `cartopy`, as a starter i.e., `cartopy/tools/feature_download.py` is not bundled within `cartopy` as a package. So really, it's a development tool.

If we're serious about packaging this with `iris`, then we could easily make an entry-point for it, then it would indeed be accessible to users via the command line on install.

I didn't want to do that work up front, only for people to object... I rather discuss it, have agreement, then do it.

**Comment by bjlittle** [comment]
So since we're on topic... what would you prefer?

**Comment by rcomer** [comment]
@bjlittle my opinion is to punt that question to a new Issue: the scope on this PR has ~creeped~ swerved far enough, and fixing the CI should be the first priority!

**Comment by trexfeathers** [comment]
> @bjlittle my opinion is to punt that question to a new Issue: the scope on this PR has creeped swerved far enough, and fixing the CI should be the first priority!

That's way better than what I was about to write. PR's that solve multiple problems are an anti-pattern, and as @rcomer eludes, this isn't exactly a small add-on.

**Comment by bjlittle** [comment]
Totally happy to follow this up with a GH discussion :+1: 

Anyone fancy reviewing and merging?

I'm keen to unblock CI

**Comment by bjlittle** [comment]
Note that https://github.com/SciTools/cartopy/pull/1602 renames `tools/feature_download.py` to be `tools/cartopy_feature_download.py` and also makes it a pucker command line script (finally) :partying_face: 

That PR was merged in May 2021 and didn't make the cut for the April release of `cartopy` 0.19.0.post1... but it will be available in the next release of `cartopy`, as will https://github.com/SciTools/cartopy/pull/1833

**Comment by pp-mo** [comment]
@rcomer @trexfeathers 
I'm good with this as it is.  I reviewed the code and its CLI and I think it works OK.
I do also agree that including this as an installed script is another step, and we don't want to get into that yet
- assuming that is what everyone else meant by "punt that question to a new Issue" / "PR's that solve multiple problems"

Going to merge unless someone objects...

**Comment by bjlittle** [comment]
Once this PR is merge, I'll make an associated issue to discuss follow-up work.

This is all in a state of flux, and we may insist on a minimum pin for `cartopy` within `iris` moving forwards to make our lives easier :thinking: ... but that's for later to decide

**Review by jamesp — CHANGES_REQUESTED** [review]
Suggest we pause on this and consider putting changes into cartopy instead of iris

**Comment by rcomer** [comment]
Oh look 👀 https://github.com/conda-forge/cartopy-feedstock/pull/116

**Comment by bjlittle** [comment]
That's only a fix to 0.19.0.post1, and not older versions of `cartopy`

**Comment by rcomer** [comment]
@bjlittle but it would get our CI moving?

File: .cirrus.yml
Comment (bjlittle) [inline .cirrus.yml:80]: Temporary, for now... pending a PR to cartopy

Comment (jamesp) [reply .cirrus.yml:80]: Nice!

**Comment by bjlittle** [comment]
@jamesp ... and finally, this is good to go :crossed_fingers:

**Review by trexfeathers — APPROVED** [review]

**Review by jamesp — APPROVED** [review]

**Pull request merged**
